src/main.rs

```diff
@@ -26,14 +26,24 @@ async fn main() {
     let pool = config.pg.create_pool(Some(Runtime::Tokio1), NoTls).unwrap();
 
     let state_filter = warp::any().map(move || pool.clone());
+    let necessity = state_filter.and(warp::header::<String>("authorization"));
```

Comment:
why is this here? `.and(…)` doenst seem all that bad and even if it was `necessity` is sub-par naming-wise

Reply:
Having to add the state filter and authorization header individually on every route is a bit annoying, however, I do agree the name is not that great, what do you think it should be named?

Reply:
auth_header i guess, tbh why not just make the two and then combine 'em and then make the combined version a filter